awx/main/scheduler/kubernetes.py

```diff
@@ -51,6 +52,72 @@ def list_active_jobs(self, instance_group):
 
         return pods
 
+    @classmethod
+    def create_secret(self, job):
+        task = collections.namedtuple('Task', 'id instance_group')(id='', instance_group=job.instance_group)
+        pm = PodManager(task)
+        registry_cred = job.execution_environment.credential
+        host = registry_cred.get_input('host').split('/')[0]
+        username = registry_cred.get_input("username")
+        password = registry_cred.get_input("password")
+
+        # Construct container auth dict and base64 encode it
+        token = b64encode("{}:{}".format(username, password).encode('ascii')).decode()
+        auth_dict = json.dumps({"auths": {host: {"auth": token}}}, indent=4)
+        auth_data = b64encode(str(auth_dict).encode('ascii')).decode()
+
+        # Construct Secret object
+        secret = client.V1Secret()
+        secret_name = "automation-{0}-image-pull-secret-{1}".format(settings.INSTALL_UUID[:5], job.execution_environment.credential.id)
+        secret.metadata = client.V1ObjectMeta(name="{}".format(secret_name))
+        secret.type = "kubernetes.io/dockerconfigjson"
+        secret.kind = "Secret"
+        secret.data = {".dockerconfigjson": auth_data}
+
+        # Check if secret already exists
+        secrets = None
+        try:
+            secrets = pm.kube_api.list_namespaced_secret(namespace=pm.namespace)
+        except client.rest.ApiException:
+            error_msg = 'Invalid openshift or k8s cluster credential'
```

Comment:
Any time we interact with a 3rd party API I think. "How can this thing fail?". Could have a DNS error, timeout, partial response, etc. So two things (1) I'm not sure how the library we are using surfaces those failures, are they all `client.rest.ApiException`? (2) It would be REALLY nice to tell users more about the failure. I do see that we have a logger.exception, that is helpful but requires digging in the logs when there is a failure. It's a balancing act though, it's tempting to just toss the entire exception into `job_explanation` but the response maybe contains sensitive information (I doubt it does, but maybe?). We should look at a few failure scenarios and see if we can do something nice for the user. For example https://github.com/ansible/awx/blob/devel/awx/api/views/__init__.py#L1677

Reply:
@chrismeyersfsu we show job_explanation in the UI now as part of the job status, so we probably don't want a super long traceback there.  

![image](https://user-images.githubusercontent.com/11698892/119364683-2275a000-bc7d-11eb-9898-e72b672506e8.png)

Reply:
After discussion, I switch these back to raising exception rather than canceling the job.  As a result, the status message in the UI will be "Error"